### PR TITLE
quic: use a utility helper for preferred address

### PR DESCRIPTION
**Pull request by jasnell**
Make working with preferred address a bit easier

### CODE REVIEW

File: src/node_quic_util-inl.h [hunk 1]
```diff
@@ -234,6 +188,101 @@ const char* QuicError::GetFamilyName() {
...
+}
+const int16_t QuicPreferredAddress::PreferredIPv4Port() const {
+  return paddr_->ipv4_port;
+}
+
+bool QuicPreferredAddress::Use(int family) const {
+  uv_getaddrinfo_t req;
+
+  if (!ResolvePreferredAddress(family, &req) ||
+      req.addrinfo == nullptr) {
```
Comment (addaleax) [inline src/node_quic_util-inl.h]: Should `uv_freeaddrinfo()` be called in the second case?

Comment (jasnell) [reply src/node_quic_util-inl.h]: No, if `ResolvePreferredAddress()` returns false, then there's no need to call `uv_freeaddrinfo`

Comment (addaleax) [reply src/node_quic_util-inl.h]: Yeah, I know, but in the `req.addrinfo == nullptr` case that isn’t quite as obvious?

Comment (jasnell) [reply src/node_quic_util-inl.h]: If req.addrinfo is nullptr, there's nothing to free.

**Comment by jasnell** [comment]
Landed

**Pull request closed without merging**